tools/src/prepare-for-vale/KeepDescriptions.ts

```diff
@@ -45,7 +45,7 @@ export default class KeepDescriptions {
       } else if (inside_text && line.match(/^[\s]*[\w\\$]*:/)) {
         inside_text = false
       } else if (inside_text) {
-        const cleaned_line = line.replace(/\[([^\]]+)\]\([^)]+\)/g, '$1')
+        const cleaned_line = this.remove_links(line)
```

Comment:
You can also have links in descriptions on line 44, so it will need to be handled there as well (and possibly in other cases). Add tests for those of course. 

We have `remove_links`, `prune_vars` and occasionally `prune`. Either extract the common parts into a method, or at least make it visible/more readable that we're doing a series of transformations.

```typescript
var cleaned_line = this.prune(line, /(description|x-deprecation-message):/, ' '))
cleaned_line = this.remove_links(line)
cleaned_line = this.prune_vars(cleaned_line)
fs.writeSync(writer, cleaned_line)
```

```typescript 
let cleaned_line = this.remove_links(line)
cleaned_line = this.prune_vars(cleaned_line)
fs.writeSync(writer, cleaned_line)
```